examples/spring-example/src/test/kotlin/com/stove/spring/example/e2e/ExampleTest.kt

```diff
@@ -133,8 +133,12 @@ class ExampleTest : FunSpec({
             }
 
             kafka {
-                publish("trendyol.stove.service.product.create.0", productCreateEvent)
-                shouldBeConsumed<ProductCreateEvent> { actual ->
+                publish("trendyol.stove.service.product.create.0", productCreateEvent, headers = mapOf("test" to "test"))
+                shouldBeConsumed<ProductCreateEvent>(metadataCondition = {
```

Comment:
Do we need metadataCondition? 
I ask this this question, because you've added metadata to condition field.
`        condition: (Option<T>, metadata: KafkaMetadata) -> Boolean
`

Reply:
Yes, they are different.

To create a nice API I need to change current API and make some breaking changes.
Instead of having 2 assertion function, I think it is better to collect them under one:

```kotlin
shouldBeConsumed<ProductCreatedEvent>{ actual: ObservedMessage<ProductCreatedEvent> ->
  actual.message.id == "123" && actual.metadata.headers.containsKey("test")
 }
```

So, we need to introduce the `ObservedMessage` data class. While doing that, we should not break the users who assert without `headers` or metadata. So, metadata assertion is always true if not explicitly asked by the user.